airbyte-webapp/src/packages/cloud/components/experiments/FreeConnectorProgram/InlineEnrollmentCallout.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { PropsWithChildren } from "react";
+import { FormattedMessage } from "react-intl";
+
+import { Callout } from "components/ui/Callout";
+import { Text } from "components/ui/Text";
+
+import { useExperiment } from "hooks/services/Experiment";
+
+import styles from "./InlineEnrollmentCallout.module.scss";
+
+const EnrollLink: React.FC<PropsWithChildren<unknown>> = ({ children }) => {
+  const onEnrollClick = () => {
+    return null;
+  };
+
+  return (
+    <span
+      role="button"
+      onClick={() => onEnrollClick()}
+      onKeyDown={(e) => e.keyCode === 13 && onEnrollClick}
+      tabIndex={0}
+    >
+      {children}
+    </span>
+  );
+};
+export const InlineEnrollmentCallout: React.FC = () => {
+  const isFreeConnectorProgramEnabled = useExperiment("workspace.freeConnectorsProgram.visible", false);
+
+  // todo: implement actual call once merged with issue #4006
+  // for now, we'll just default to true
+  const enrolledInFreeConnectorProgram = false;
```

Comment:
Open to this logic and the logic above it living instead in the place this is implemented.

My reason for handling this way is that it makes it easier to remove or change all the related code later.  I can see the argument the other way too, not strongly tied to this.